pkg/controller/kfservice/reconcilers/knative/configuration_reconciler.go

```diff
@@ -104,6 +111,26 @@ func (r *ConfigurationReconciler) reconcileCanary(kfsvc *v1alpha1.KFService) err
 	return nil
 }
 
+func (r *ConfigurationReconciler) finalizeConfiguration(kfsvc *v1alpha1.KFService) error {
+	canaryConfigurationName := constants.CanaryConfigurationName(kfsvc.Name)
+	existing := &knservingv1alpha1.Configuration{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: canaryConfigurationName, Namespace: kfsvc.Namespace}, existing)
```

Comment:
Better style to inline the error?

if err := myFunc(); err != nil {
   somethingWith(err)
}

Same below with the delete call.